### PR TITLE
feat: two-factor authentication - verify

### PR DESCRIPTION
**Pull request by scottandee**
## What does this Pull Request Do?
This PR is about verifying a two factor authentication code for additional layer of security.

## How should this be manually tested?
As an authenticated user, send a POST request to `/api/v1/auth/2fa/verify` with a request body as below:
```bash
{
    "totp_code": "123456"
}
```
## Checklist of what you did
- [x] Implemented endpoint to verify the TOTP for 2fa authentication.
- [x] Generate and store backup codes
- [x] Returned success message with the right response payload.
- [x] Added unit tests for the auth service

Issue Link: [Issue 266](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/266)

### CODE REVIEW

**Review by rugue — APPROVED** [review]

**Review by rugue — DISMISSED** [review]
ci cd test successful

**Comment by scottandee** [comment]
@buka4rill awaiting your review

File: src/modules/user/user.service.ts [hunk 1]
```diff
@@ -44,6 +46,10 @@ export default class UserService {
     return await GetRecord[identifierType]();
   }
 
+  async saveUser(user: Partial<UserInterface>) {
```
Comment (Homoakin619) [inline src/modules/user/user.service.ts:49]: I actually have some reservations on this method. Does this method not create a new user ? If yes then there is an existing method for that in the user service

Comment (scottandee) [reply src/modules/user/user.service.ts:49]: It doesn't create a new user. It just calls the save method on the user that I retrieved in the 2fa method. The verify2fa method needs to update the two_factor secret and backup_codes records on the user's table.
An alternative method which I did not have at the time of writing the code is the pass the updates into the updateUser method. @Homoakin619

Comment (scottandee) [reply src/modules/user/user.service.ts]: I've removed that method and made use of the updateUserRecord

**Review by Heba-WebDev — DISMISSED** [review]

**Comment by Heba-WebDev** [comment]
2fa no longer supported

**Review by Heba-WebDev — APPROVED** [review]

**Review by rugue — APPROVED** [review]

**Review by Homoakin619 — APPROVED** [review]

**Pull request merged**